### PR TITLE
Task add onComplete and onError

### PR DESCRIPTION
**Pull request by e111077**
Adds `onComplete` and `onError` callbacks to Task.

### CODE REVIEW

**Comment by e111077** [comment]
Ping on this one. Would like to get it merged before talk

File: packages/labs/task/src/task.ts [hunk 1]
```diff
@@ -215,9 +221,11 @@ export class Task<
         if (error === undefined) {
           this.status = TaskStatus.COMPLETE;
           this._resolveTaskComplete(result as R);
+          this._onComplete?.(result as R);
```
Comment (justinfagnani) [inline packages/labs/task/src/task.ts:224]: If we ever wanted onComplete to be able to change completion status by returning something, we'd want to run this first.

Comment (justinfagnani) [inline packages/labs/task/src/task.ts:224]: I think this needs to be in a try/catch block to keep from messing with task state if it throws. Same for onError.

**Review by justinfagnani — APPROVED** [review]

File: packages/labs/task/src/task.ts (hunk 1 above)
Comment (e111077) [reply packages/labs/task/src/task.ts:224]: what should we do on catch? It's already too late to report onError, no?

Comment (e111077) [reply packages/labs/task/src/task.ts:224]: Something like this?

```ts
if (error === undefined) {
  try {
    this._onComplete?.(result as R);
  } catch {
    // Ignore user errors from onComplete.
  }
  this.status = TaskStatus.COMPLETE;
  this._resolveTaskComplete(result as R);
} else {
  try {
    this._onError?.(error);
  } catch {
    // Ignore user errors from onError.
  }
  this.status = TaskStatus.ERROR;
  this._rejectTaskComplete(error);
}
this._value = result as R;
this._error = error;
 ```

Comment (justinfagnani) [reply packages/labs/task/src/task.ts:224]: exactly

**Pull request merged**